HHH-17612 DefaultRevisionEntity: Illegal argument on static metamodel field injection

mbladel opened the pull request:
<!--
If this is your first time contributing to the project, please consider reviewing https://github.com/hibernate/hibernate-orm/blob/main/CONTRIBUTING.md
-->

https://hibernate.atlassian.net/browse/HHH-17612

Alternative solution to https://github.com/hibernate/hibernate-orm/pull/9027, instead of creating new mappings for default revision entities we're creating new `@MappedSuperclasses` types which can be extended by both Envers and user's revision entities. Suggestions on the naming of these types are appreciated.

I also added to the migration guide and cleaned up the Envers user guide chapters about revision entities.

<!--
Please read and do not remove the following lines:
-->
----------------------
By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt)
and can be relicensed under the terms of the [LGPL v2.1 license](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.txt) in the future at the maintainers' discretion.
For more information on licensing, please check [here](https://github.com/hibernate/hibernate-orm/blob/main/CONTRIBUTING.md#legal).

----------------------

sebersole commented:
I'm not sure `DefaultRevisionType` is the default of anything.  It seems to me the types here ought to be something like:

```java
@MappedSuperclass
public class RevisionMapping {
    ...
}

@Entity
class DefaultRevisionEntity extends RevisionMapping {
    ...
}
```

mbladel commented:
Sure, so I've removed "default" for the two mapped-superclasses that had it in the name, and replaced "type" with "mapping" for all of them. LMK if there's anything else I should change.

sebersole reviewed (approved):
lgtm
